### PR TITLE
feat: Adds `last_k` parameter to `ChatMessageRetriever` init/run methods

### PR DESCRIPTION
**Pull request by vblagoje**
### Why:
Enhances the `ChatMessageRetriever` component by introducing a `last_k` parameter allowing for selective retrieval of a specified number of recent chat messages. This change addresses the need for more control in accessing chat history, particularly in scenarios where only the most recent interactions are relevant.

- fixes https://github.com/deepset-ai/haystack/issues/8258

### What:
- Added `last_k` as init parameter to match the existing pattern in other retrievers (defaults to 10)
- Added an `Optional[int]` parameter named `last_k` to the `run` method of the `ChatMessageRetriever`, enabling selection of the number of recent messages to retrieve.
- Implemented validation to ensure `last_k` is greater than 0 when specified, raising a `ValueError` if not.
- Expanded unit tests to cover new functionalities, including cases where `last_k` is specified, and edge cases where `last_k` is set to 0 or a negative number, ensuring robustness through comprehensive testing.

### How can it be used:
```python
from haystack import Pipeline
from haystack.components.builders import ChatPromptBuilder
from haystack.dataclasses import ChatMessage

from haystack_experimental.components.retrievers import ChatMessageRetriever
from haystack_experimental.chat_message_stores.in_memory import InMemoryChatMessageStore

messages = [
            ChatMessage.from_user(content="Hello, how can I help you?"),
            ChatMessage.from_user(content="Hallo, wie kann ich Ihnen helfen?"),
            ChatMessage.from_user(content="Hola, como puedo ayudarte?"),
            ChatMessage.from_user(content="Bonjour, comment puis-je vous aider?")
        ]

        message_store = InMemoryChatMessageStore()
        message_store.write_messages(messages)
        retriever = ChatMessageRetriever(message_store)

assert retriever.run(last_k=2) == {
            "messages": [ChatMessage.from_user(content="Hola, como puedo ayudarte?"),
                         ChatMessage.from_user(content="Bonjour, comment puis-je vous aider?")
                         ]}
```
### How did you test it:
- Unit tests were written to verify the correct retrieval of messages when a valid `last_k` value is provided.
- Tests ensure correct behavior when `last_k` exceeds the number of available messages, in which case all messages should be returned.
- Edge cases where `last_k` is set to 0 or a negative number were tested to confirm that a `ValueError` is appropriately raised.

### Notes for the reviewer:
- We first used `top_k` as the name for this parameter but have since renamed it to `last_k`. See https://github.com/deepset-ai/haystack/issues/8258 for further discussion

### CODE REVIEW

File: haystack_experimental/components/retrievers/chat_message_retriever.py [hunk 1]
```diff
@@ -39,14 +39,19 @@ class ChatMessageRetriever:
...
 
         :param message_store:
             An instance of a ChatMessageStore.
+        :param last_k:
+            The number of last messages to retrieve. Defaults to 10 messages if not specified.
         """
         self.message_store = message_store
+        if last_k <= 0:
+            raise ValueError(f"last_k must be greater than 0. Currently, the last_k is {last_k}")
+        self.last_k = last_k
```
Comment (Amnah199) [inline haystack_experimental/components/retrievers/chat_message_retriever.py:54]: Can there be cases where the user might want to set `last_k=0` to not retrieve any chat messages in that particular run or thats completely invalid? Just a thought 🤔

Comment (vblagoje) [reply haystack_experimental/components/retrievers/chat_message_retriever.py:54]: Hm, good point. Perhaps there could be cases like the one you describe. I'll adjust this check 🙏

**Review by Amnah199 — APPROVED** [review]
Looks good!

**Comment by vblagoje** [comment]
@Amnah199 after internal consult with @shadeMe we decided to drop the last commit where allowed 0 to be a valid last_k value. We can always adjust it if needed for these theoretically possible long tail use cases.

**Review by shadeMe — APPROVED** [review]
